src/components/forms/DatePicker/utils.tsx

```diff
@@ -0,0 +1,522 @@
+import React from 'react'
+
+import { DEFAULT_EXTERNAL_DATE_FORMAT, INTERNAL_DATE_FORMAT } from './constants'
+
+/**
+ * This file contains the USWDS DatePicker date manipulation functions converted to TypeScript
+ */
```

Comment:
So I get that we are just doing this for speed and copy/pasting USWDS tests but I'm a bit weirded out that this doesn't have 100% unit tests coverage for every function for maintenance reasons. Maybe we should backlog a story to either test everything here or else swap with an external dependency (some date/time lib) that would already be well tested?

Reply:
USWDS didn't seem to have tests for these methods at all. I started writing some but I haven't gotten them all.. I'm hoping to get as much coverage on these as possible but some of them are just tricky because dates

Reply:
Oh wow. 🥇 Okay well then thank you for starting on this then, I misunderstood!  Do you not want to lean on a lightweight date/time lib or util? I can see why we wouldn't want to do that just for this component... but just checking

Reply:
yeah I thought about it, but it would mean importing a new lib and I'm making the assumption that USWDS wrote their own utils to keep things lightweight. I think I want to just follow their decision on this.

Reply:
this also occurred to me later on, but if USWDS makes updates to their DatePicker code then it will be harder to reconcile differences if we deviate from them.